src/GraphQL/DI/IConfigureSchema.cs

```diff
@@ -0,0 +1,31 @@
+using System;
+using GraphQL.Types;
+
+namespace GraphQL.DI
+{
+    /// <summary>
+    /// Allows configuration of a schema prior to its constructor.
+    /// </summary>
+    public interface IConfigureSchema
+    {
+        /// <summary>
+        /// Configures a schema prior to the code in its constructor.
+        /// </summary>
+        void Configure(ISchema schema, IServiceProvider serviceProvider);
+    }
+
+    internal class ConfigureSchema : IConfigureSchema
+    {
+        private readonly Action<ISchema, IServiceProvider> _action;
+
+        public ConfigureSchema(Action<ISchema, IServiceProvider> action)
+        {
+            _action = action ?? throw new ArgumentNullException(nameof(action));
+        }
+
+        public void Configure(ISchema schema, IServiceProvider serviceProvider)
```

Comment:
By the way, `serviceProvider` here is the root service provider, not scoped one even if schema is scoped. MSDI always injects root service provider. When I wrote SteroidsDI, I was faced with this non-obvious problem.

Reply:
Really? I suppose I always assumed that options configurations were singletons which would naturally be the root service provider.

Reply:
It *may* be a problem, may be not.